tools/dependency/pip_check.py

```diff
@@ -0,0 +1,94 @@
+#!/usr/bin/python3
+
+import os
+import sys
+from functools import cached_property
+
+import yaml
+
+DEPENDABOT_CONFIG = ".github/dependabot.yml"
+
+# TODO(phlax): add checks for:
+#      - requirements can be installed together
+#      - pip-compile formatting (maybe do that in code_format)
+
+
+# TODO(phlax): move this to a base module
+class Checker(object):
```

Comment:
@phlax can this verify that we're doing pinning for all `requirements.txt`? CC @moderation

Reply:
not yet, but yes, kinda in 2 ways

we could add pip-compile which is kinda like pipfmt, and can check/generate hashes

secondly we could/should run a bazel query to ensure that all of the pip targets use `require-hashes`

Reply:
can we do this in a follow up, it brings quite a few considerations beyond this PR